### PR TITLE
[Performance Optimization] Improve the speed of writing index file in shuffle server

### PR DESCRIPTION
**Pull request by zuston**
### What changes were proposed in this pull request?
[Performance Optimization] Improve the speed of writing index file in shuffle server

### Why are the changes needed?
When I test uniffle performance, i found a huge performance drop due to the low speed of writing index file. 

After using "dataOutputStream = new DataOutputStream(new BufferedOutputStream(fileOutputStream))" in LocalFileWriter, the performance has great imrpovement.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
No need.

### CODE REVIEW

**Comment by jerqi** [comment]
Could you modify the description according to our pull request template? We will use the description as commit message,  the commit message will provide more information to help people understand the code.

File: storage/src/main/java/org/apache/uniffle/storage/handler/impl/LocalFileWriter.java [hunk 1]
```diff
@@ -34,7 +35,7 @@ public class LocalFileWriter implements Closeable {
   public LocalFileWriter(File file) throws IOException {
     fileOutputStream = new FileOutputStream(file, true);
     // init fsDataOutputStream
-    dataOutputStream = new DataOutputStream(fileOutputStream);
+    dataOutputStream = new DataOutputStream(new BufferedOutputStream(fileOutputStream));
```
Comment (jerqi) [inline storage/src/main/java/org/apache/uniffle/storage/handler/impl/LocalFileWriter.java:38]: Will BufferedOutputStream influence the performance of writing data file? Will it increase one time data copy when we use BufferedOutputStream?

Comment (zuston) [reply storage/src/main/java/org/apache/uniffle/storage/handler/impl/LocalFileWriter.java:38]: From the flamegraph, i dont found this problem. But for the rule of minor change, i think we should make this change only for write index. What do u think?

Comment (jerqi) [reply storage/src/main/java/org/apache/uniffle/storage/handler/impl/LocalFileWriter.java:38]: I'm a little worried that the data copy will increase gc time.

Comment (jerqi) [reply storage/src/main/java/org/apache/uniffle/storage/handler/impl/LocalFileWriter.java:38]: cc @colinmjj

Comment (colinmjj) [reply storage/src/main/java/org/apache/uniffle/storage/handler/impl/LocalFileWriter.java:38]: Compare with shuffle data, the size of index data can be ignored. And gc shouldn't be a problem with this improvement.

Comment (jerqi) [reply storage/src/main/java/org/apache/uniffle/storage/handler/impl/LocalFileWriter.java:38]: @colinmjj You're right
BufferedOutpuStream's implement is as below
```
    public synchronized void write(byte b[], int off, int len) throws IOException {
        if (len >= buf.length) {
            /* If the request length exceeds the size of the output buffer,
               flush the output buffer and then write the data directly.
               In this way buffered streams will cascade harmlessly. */
            flushBuffer();
            out.write(b, off, len);
            return;
        }
        if (len > buf.length - count) {
            flushBuffer();
        }
        System.arraycopy(b, off, buf, count, len);
        count += len;
    }
```
When data is big enough, we won't copy data to buffer.
+1 for this PR

**Review by jerqi — APPROVED** [review]
Good catch! LGTM, thanks @zuston

**Pull request merged**